compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp

```diff
@@ -103,28 +105,48 @@ makeEncoding(OpBuilder &builder, IREE::LinalgExt::EncodingUser user,
       context, userAttr, roleAttr, operandElemTypesAttr, originalTypeAttr);
 }
 
-static Value padAndSetEncoding(OpBuilder &builder, Location loc, Value source,
-                               IREE::LinalgExt::EncodingUser user,
-                               IREE::LinalgExt::EncodingRole role,
-                               TypeRange operandTypes) {
+static Value castEncodedResult(OpBuilder &builder, Location loc, Value encoded,
+                               Value castResult, CastOpInterface castOp,
+                               IREE::LinalgExt::EncodingAttr encodingAttr) {
+  auto encodedType = cast<RankedTensorType>(encoded.getType());
+  auto castResultType = cast<RankedTensorType>(castResult.getType());
```

Comment:
do we need `castResult` being passed? Can we get it from `getElementTypeOrSelf(castOp->getResultTypes()[0])`?